tests/core/engine_adapter/test_integration.py

```diff
@@ -872,3 +881,74 @@ def test_scd_type_2(ctx: TestContext):
             ]
         ),
     )
+
+
+def test_sushi(ctx: TestContext):
+    if ctx.test_type == "df":
+        pytest.skip("Sushi end-to-end tests only need to run for query and pyspark tests")
+
+    # this skips the pyspark test if the engine isn't pyspark-enabled
+    ctx.init()
+
+    config = load_config_from_paths(
+        project_paths=[
+            pathlib.Path(os.path.join(os.path.dirname(__file__), "config.yaml")),
+        ],
+        personal_paths=[pathlib.Path("~/.sqlmesh/config.yaml").expanduser()],
+    )
+    gateway = "inttest_mssql" if ctx.dialect == "tsql" else f"inttest_{ctx.dialect}"
+    context = Context(paths="./examples/sushi", config=config, gateway=gateway)
+
+    start = to_date(now() - timedelta(days=7))
+    end = now()
+
+    context.plan(
+        environment="test_prod",
+        start=start,
+        end=end,
+        skip_tests=True,
+        no_prompts=True,
+        auto_apply=True,
+    )
+
+    data_validator = SushiDataValidator.from_context(context)
```

Comment:
Thanks for including this! I look forward to when we have additional validations for other tables.

Reply:
That can be a later PR. I think making sure we validate the data for each model that has a unique load pattern would be good. I wonder if we should consider freezing in order to hard-code expected values. We can figure this out later though.